### ISSUE TITLE
Bug: [no-invalid-void-type] false positive with `Promise.withResolvers<void>()`

### ISSUE THREAD

**Issue by AlexCannonball** [typescript-eslint/typescript-eslint#12040]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.3&fileType=.ts&code=MYewdgzgLgBA3jADgJxAWwJYQKYBobLYBW2wU%2BhEIANgG7YwC%2BMAvDAAqqY4B0A7higALAErYqdbMggAeWiAwATAHwAKAJQBuALAAoIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1uYDcBDRgBNaPDpWFFS6KImjQO0SODABfECqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkqyARhWANrg23okBMADNwBZIAGnYdIXIugAeeAHQBDTJgUBPEWJoSSU2XLioM0AsQ0dOOmfOR4sCvLEyRNAXVHj4sACYBXDM0gib3RdPDMtPCUAc3Qwym0%2Bbm5w2hI8TGRjZnSfdHctLgBZbz90AEkiW0xYRGzMXPzUgGtkRAAZRgBhAAt0aCa6hs1IT18MACUuWFQfEyIAoK90AEYANmcIAF8QTaA&tokens=false

### Repro Code

```TypeScript
const { promise, reject, resolve } = Promise.withResolvers<void>();
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-invalid-void-type": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "lib": [
      "es2024",
      "esnext.array",
      "esnext.collection",
      "esnext.iterator"
    ],
    "module": "nodenext",
    "target": "es2022",
    "strict": true,
    "esModuleInterop": true,
    "skipLibCheck": true,
    "moduleResolution": "node16"
  }
}
```

### Expected Result

There should be no error `no-invalid-void-type` for `Promise.withResolvers<void>()`.

### Actual Result

`Promise.withResolvers<void>()` is flagged:

> void is only valid as a return type or generic type argument. 1:60 - 1:64

### Additional Info

Hello,

I guess, `Promise.withResolvers()` was introduced in ES2024. Node.js 22 should support ES2024. So I've used https://github.com/tsconfig/bases/blob/main/bases/node22.json to fill `compilerOptions`.

Playing with `lib` (like, setting to `"esnext"`) didn't remove the error.

**Comment by kirkwaiblinger** [comment]
See https://github.com/typescript-eslint/typescript-eslint/issues/9421
